rskj-core/src/main/java/co/rsk/validators/BlockTimeStampValidationRule.java

```diff
@@ -18,22 +18,51 @@
 
 package co.rsk.validators;
 
+import co.rsk.bitcoinj.core.BtcBlock;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.params.RegTestParams;
+import co.rsk.util.TimeProvider;
+import org.ethereum.config.Constants;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
 import org.ethereum.core.Block;
 import org.ethereum.core.BlockHeader;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
 /**
  * Created by mario on 23/01/17.
  */
 public class BlockTimeStampValidationRule implements BlockParentDependantValidationRule, BlockHeaderParentDependantValidationRule, BlockValidationRule, BlockHeaderValidationRule {
 
     private static final Logger logger = LoggerFactory.getLogger("blockvalidator");
 
-    private int validPeriodLength;
+    private static final long MAX_TIMESTAMPS_DIFF_IN_SECS = Constants.getMaxTimestampsDiffInSecs();
+
+    private final int validPeriodLength;
+    private final ActivationConfig activationConfig;
+    private final TimeProvider timeProvider;
+    private final NetworkParameters bitcoinNetworkParameters;
 
-    public BlockTimeStampValidationRule(int validPeriodLength) {
+    public BlockTimeStampValidationRule(int validPeriodLength, ActivationConfig activationConfig,
+                                        TimeProvider timeProvider, NetworkParameters bitcoinNetworkParameters) {
         this.validPeriodLength = validPeriodLength;
+        this.activationConfig = Objects.requireNonNull(activationConfig);
+        this.timeProvider = Objects.requireNonNull(timeProvider);
+        this.bitcoinNetworkParameters = Objects.requireNonNull(bitcoinNetworkParameters);
+    }
+
+    public BlockTimeStampValidationRule(int validPeriodLength, ActivationConfig activationConfig,
+                                        TimeProvider timeProvider) {
+        this(validPeriodLength, activationConfig, timeProvider, RegTestParams.get());
```

Comment:
Its the same to use RegTestParams than MainNetParams in this logic class? Because I dont see any other use than this default constructor, so no matter which network we are, it will use regtest params.

Reply:
Yes, this is bitcoin network params and only this `RegTestParams` is being used across our code base.